std-bits/microsoft/src/main/java/org/enso/microsoft/azure/AzureBlobStorage.java

```diff
@@ -0,0 +1,89 @@
+package org.enso.microsoft.azure;
+
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.slf4j.Logger;
+
+public final class AzureBlobStorage {
+  private static final Logger LOGGER = org.slf4j.LoggerFactory.getLogger(AzureBlobStorage.class);
+
+  public static List<String> containers(
+      AzureCredential credential, String storageAccountName, String prefix) {
+    LOGGER.warn("Reading from Blob Storage: {}", storageAccountName);
+
+    var client =
+        new BlobServiceClientBuilder()
+            .endpoint("https://" + storageAccountName + ".blob.core.windows.net/")
+            .credential(CredentialHelper.toTokenCredential(credential))
+            .buildClient();
+
+    var containerList = client.listBlobContainers();
+
+    var result = new ArrayList<String>();
+    for (var blobContainer : containerList) {
+      var containerName = blobContainer.getName();
+      if ((Objects.equals(prefix, "")) || containerName.startsWith(prefix)) {
+        result.add(containerName);
+      }
+    }
+    return result;
```

Comment:
Stylistic suggestion. Feel free to ignore.

```suggestion
    return containerList.stream()
        .map(BlobContainerItem::getName)
        .filter(name -> prefix.isEmpty() || name.startsWith(prefix))
        .collect(Collectors.toList());
```